hera_mc/geo_handling.py

```diff
@@ -19,298 +21,397 @@
-                                         'station_type': stn.station_type_name,
-                                         'longitude': stn.lon,
-                                         'latitude': stn.lat,
-                                         'elevation': stn.elevation,
-                                         'antenna_number': ant_num,
-                                         'start_date': start_date,
-                                         'stop_date': stop_date})
-    return stations_new
-
-
-def get_since_date(args,query_date):
-    dt = query_date.gps
-    db = mc.connect_to_mc_db(args)
-    found_stations = []
-    with db.sessionmaker() as session:
-        for a in session.query(geo_location.GeoLocation).filter(geo_location.GeoLocation.created_gpstime >= dt):
-            found_stations.append(a.station_name)
-    return found_stations
-
-coord = {'E': 'easting', 'N': 'northing', 'Z': 'elevation'}
-
-
-def plot_stations(args, stations_to_plot, fignm, query_date=None, marker_color='g', marker_shape='o', marker_size='8', label_station=False):
-    """Plot a list of stations.
-
-       Parameters:
-       ------------
-       args:  needed arguments to open database and set date/time
-       stations_to_plot:  list containing station_names (note:  NOT antenna_numbers)
-       fignm:  figure name/number to plot on
-       marker_color, marker_shape:  color and shape of marker
-       label_station:  flag to either label the station or not.  If not false, give label type.
-    """
-
-    plt.figure(fignm)
-    db = mc.connect_to_mc_db(args)
-    with db.sessionmaker() as session:
+                    stations_conn.append({'station_name': stn.station_name,
+                                          'station_type': stn.station_type_name,
+                                          'longitude': stn.lon,
+                                          'latitude': stn.lat,
+                                          'elevation': stn.elevation,
+                                          'antenna_number': ant_num,
+                                          'start_date': start_date,
+                                          'stop_date': stop_date})
+        return stations_conn
+
+
+    def get_since_date(self,query_date,station_types_to_check):
+        self.get_station_types(add_stations=False)
+        dt = query_date.gps
+        found_stations = []
+        for a in self.session.query(geo_location.GeoLocation).filter(geo_location.GeoLocation.created_gpstime >= dt):
+            if self.flipped_station_types[a.station_type_name] in station_types_to_check:
+                found_stations.append(a.station_name)
+        return found_stations
+
+
+    def plot_stations(self, stations_to_plot, query_date, state_args):
+        """
+        Plot a list of stations.
+
+        Parameters:
+        ------------
+        stations_to_plot:  list containing station_names (note:  NOT antenna_numbers)
+        query_date:  date for active-only plot
```

Comment:
this description suggests that query_date is optional if you don't want an active-only plot, but since it's positional it's actually required.